python/pyspark/worker.py

```diff
@@ -500,7 +501,12 @@ def main(infile, outfile):
 
             except (resource.error, OSError, ValueError) as e:
                 # not all systems support resource limits, so warn instead of failing
-                print("WARN: Failed to set memory limit: {0}\n".format(e), file=sys.stderr)
+                print(warnings.formatwarning(
+                    "Failed to set memory limit: {0}\n".format(e),
+                    ResourceWarning,
+                    __file__,
+                    getframeinfo(currentframe()).lineno
+                ))
```

Comment:
We need `file=sys.stderr`?

Reply:
We would for `warnings.showwarning` (I tested this solution as well), but here I went with `warnings.formatwarning` (which returns `str`) combined with `print` (as suggested by @HyukjinKwon).

Reply:
I think @ueshin meant to keep `file=sys.stderr` as an argument given to `print`  in order to redirect the output to standard error instead of standard out.

Reply:
Oh, my bad. Let me fix that.